src/components/Charts/chart.types.ts

```diff
@@ -0,0 +1,48 @@
+import { theme } from '@/theme';
+
+export enum ChartType {
+  Line = 'line',
+  Donut = 'donut',
+}
+export type CurveType = 'linear' | 'natural' | 'monotone' | 'step';
+
+export interface LinePoint {
+  values: Record<string, number>;
+  curveType?: CurveType | undefined;
+}
+
+export interface LineData {
+  xAxisKeys: string[];
+  lineValues: LinePoint[];
+}
+
+export interface DonutData {
+  name: string;
+  value: number;
+}
+
+export interface ChartDataMap {
+  [ChartType.Line]: LineData;
+  [ChartType.Donut]: DonutData[];
+}
+
+export interface ChartProps<K extends ChartType> {
+  type: K;
+  data: ChartDataMap[K];
+}
+
+export interface DonutProps {
+  data: DonutData[];
+  withToolTip?: boolean;
+}
+export interface LineProps {
+  data: LineData;
+}
+export const CHART_COLORS = [

```

Comment:
We dont need this object if you can simply use the theme colors inside the relevant places

Reply:
You can take this object and place it as it inside the chart component, this will be the default color schema for the chart.